pulp_2to3_migration/app/plugin/deb/pulp_2to3_models.py

```diff
@@ -184,3 +190,307 @@ def create_pulp3_content(self):
+    content creation.
+    """
+    pulp2_type = 'deb_component'
+    distribution = django_models.TextField()
+    codename = django_models.TextField()
+    component = django_models.TextField()
+    suite = django_models.TextField()
+
+    class Meta:
+        default_related_name = 'deb_component_detail_model'
+        # TODO: These are probably not the uniqueness constraints I want...
+        unique_together = ('component', 'distribution')
+
+    @classmethod
+    def pre_migrate_content_detail(cls, content_batch):
+        """
+        Pre-migrate Pulp 2 content with all the fields needed to create a Pulp 3 Content
+
+        Note that this pre_migrated_content_detail method will not just create
+        Pulp2DebComponent entries, but also Pulp2DebComponentPackage entries,
+        and Pulp2DebReleaseArchitecture entries, since they are all created from the same
+        Pulp 2 type (DebComponent).
+
+        For each Pulp2DebComponentPackage and each Pulp2DebComponent there will also be an
+        additional Pulp2Content created. Such Pulp2Content will have the same pulp2_id as
+        the original, but with a different pulp2_subid. That way we can record the one to
+        many Pulp 2 to Pulp 3 mapping later.
+
+        Args:
+             content_batch(list of Pulp2Content): pre-migrated generic data for Pulp 2 content.
+        """
+        pulp2_unit_map = {pulp2unit.pulp2_id: pulp2unit for pulp2unit in content_batch}
+        pulp2_ids = pulp2_unit_map.keys()
+        pulp2_unit_batch = pulp2_models.DebComponent.objects.filter(id__in=pulp2_ids)
+        component_units_to_save = []
+        component_package_units_to_save = []
+        release_architecture_units_to_save = []
+        pulp2_sub_records_to_save = []
+
+        for component_unit in pulp2_unit_batch:
+            pulp2_base_record = pulp2_unit_map[component_unit.id]
+            distribution = component_unit.distribution
+            codename = component_unit.release
+            component = component_unit.name
+            release = pulp2_models.DebRelease.objects.filter(
+                repoid=component_unit.repoid,
+                distribution=distribution,
+            ).first()
+            suite = release.suite
+            component_units_to_save.append(Pulp2DebComponent(
+                distribution=distribution,
+                codename=codename,
+                component=component,
+                suite=suite,
+                pulp2content=pulp2_base_record,
+            ))
+            architectures = set()
+            for package_id in component_unit.packages:
+                package_unit = pulp2_models.DebPackage.objects.filter(id__in=[package_id]).first()
+                package_relative_path = package_unit.filename
+                package_sha256 = package_unit.checksum
+                # TODO: Consider hashing the subid string to ensure it will fit in the pulp2_subid
```

Comment:
I need a second opinion on this. Does this make sense?

Reply:
I am basically ensuring that my `pulp2_subid`'s are unique by concatenating all the fields from my "unique together" conditions into a single string and using that. I am a bit worried that the concatenated strings might exceed the `max_length=255` property of the `pulp2_subid` field.

Reply:
+1 on hashing it

Reply:
Done. I also got rid of my remaining `TODO` comments.